src/verify.cc

```diff
@@ -248,9 +238,13 @@ Status verifyJwtWithoutTimeChecking(const Jwt& jwt, const Jwks& jwks) {
         return Status::Ok;
       }
     } else if (jwk->kty_ == "OKP" && jwk->crv_ == "Ed25519") {
-      if (verifySignatureEd25519(jwk->okp_key_raw_, jwt.signature_,
-                                 signed_data)) {
-        return Status::Ok;
+      Status status = verifySignatureEd25519(jwk->okp_key_raw_, jwt.signature_,
+                                             signed_data);
+      // For verification failures keep going and try the rest of the keys in
+      // the JWKS. Otherwise status is either OK or an error with the JWT and we
+      // can return immediately.
+      if (status != Status::JwtVerificationFail) {
```

Comment:
This is not easy to read.   prefer to list out the status_code that we can bail out of the loop

if (status == OK || status == WRONG_LENGTH)

Reply:
Sure. I worried about it being fragile with a list of specific statuses, but I see the flip side. Swapped it.